packages/kbn-std/src/iteration/map.ts

```diff
@@ -58,5 +58,8 @@ export async function asyncMapWithLimit<T1, T2>(
 
   return results
     .sort(([a], [b]) => a - b)
-    .reduce((acc: T2[], [, result]) => acc.concat(result), []);
+    .reduce((acc: T2[], [, result]) => {
+      acc.push(...result);
+      return acc;
+    }, []);
```

Comment:
nit: I wonder if `flatMap` would do the same...

hmmm, nevermind:
<img width="1022" alt="image" src="https://github.com/elastic/kibana/assets/5469006/694f75ad-ef56-4cc2-8224-cc1cd1cfb886">

Reply:
Hold on... that was on Chrome...

Safari 
<img width="995" alt="image" src="https://github.com/elastic/kibana/assets/5469006/39a73c51-1236-4986-99e2-e563133f50c8">

And Firefox
<img width="1003" alt="image" src="https://github.com/elastic/kibana/assets/5469006/41e0bbd0-d57b-4102-87a7-8c1bfaf83e3b">

Are consistently faster 🤷

Reply:
Running a benchmark on NodeJS:
```
flatMap x 1,226,284 ops/sec ±0.40% (97 runs sampled)
reduce with concat x 6,033,030 ops/sec ±0.19% (101 runs sampled)
reduce with push x 16,624,071 ops/sec ±0.36% (93 runs sampled)
Fastest is reduce with push
```

This is the piece of code I ran:
```JS
const Benchmark = require('benchmark');

const suite = new Benchmark.Suite;

const results = [
  [0, ["a", "b", "c"]],
  [2, ["g", "h", "i"]],
  [1, ["d", "e", "f"]],
].sort(([a], [b]) => a - b);

suite
  .add('flatMap', function () {
    const out1 = results.flatMap(([, result]) => result);
  })
  .add('reduce with concat', function () {
    const out1 = results.reduce((acc, [, result]) => acc.concat(result), []);
  })
  .add('reduce with push', function () {
    const out1 = results.reduce((acc, [, result]) => {
      acc.push(...result);
      return acc;
    }, []);
  });

suite
  // add listeners
  .on('cycle', function (event) {
    console.log(String(event.target));
  })
  .on('complete', function () {
    console.log('Fastest is ' + this.filter('fastest').map('name'));
  });

suite.run();
```